app/services/food_service.rb

```diff
@@ -0,0 +1,19 @@
+class FoodService
+  def self.foods_by_ingredient(ingredient)
+    ingredient_search = ingredient.gsub(/\s+/, "")
+    get_data("/v1/foods/search?query=#{ingredient_search}")
```

Comment:
Faraday can be a little picky about how we pass the url. If you remove the first forward slash, this should work and you won't get a 404 anymore:
`get_data("v1/foods/search?query=#{ingredient_search}")`